samples/excel/89-preview-apis/range-hyperlink.yaml

```diff
@@ -10,121 +10,267 @@ script:
+
+                let productsRange = sheet.getRange("A3:A5");
+                productsRange.load("values");
+
+                await context.sync();
+
+                // Create a hyperlink to a URL 
+                // for each product name in the first table.
+                for (let i = 0; i < productsRange.values.length; i++) {
+                    let cellRange = productsRange.getCell(i, 0);
+                    let cellText = productsRange.values[i][0];
+
+                    let hyperlink = {
+                        textToDisplay: cellText,
+                        screenTip: "Search Bing for '" + cellText + "'",
+                        address: "https://www.bing.com?q=" + cellText
+                    }
+                    cellRange.hyperlink = hyperlink;
                 }
-                range.hyperlink = hyperlink;
 
                 await context.sync();
+
+                OfficeHelpers.UI.notify("Created a hyperlink to a URL for each of the products in the first table.");
             });
         }
 
-        async function createDocumentHyperlink() {
+        async function createDocumentHyperlinks() {
             await Excel.run(async (context) => {
-                const sheet = context.workbook.worksheets.getItem("Sample");
-                const range = sheet.getRange("A7:E11");
-
-                // Create a hyperlink to a location within the document.
-                // textToDisplay and screenTip are optional attributes.
-                const hyperlink = {
-                    textToDisplay: "This links to a location in the document",
-                    screenTip: "This is the screen tip",
-                    documentReference: "A14",
+                const sheet = context.workbook.worksheets.getItem("Orders");
+
+                let productsRange = sheet.getRange("A9:A11");
+                productsRange.load("values");
+
+                await context.sync();
+
+                // Create a hyperlink to a location within the workbook
+                // for each product name in the second table.
+                for (let i = 0; i < productsRange.values.length; i++) {
+                    let cellRange = productsRange.getCell(i, 0);
+                    let cellText = productsRange.values[i][0];
+
+                    let hyperlink = {
+                        textToDisplay: cellText,
+                        screenTip: "Navigate to the '" + cellText + "' worksheet",
+                        documentReference: cellText + "!A1"
+                    }
+                    cellRange.hyperlink = hyperlink;
                 }
-                range.hyperlink = hyperlink;
 
                 await context.sync();
+
+                OfficeHelpers.UI.notify("Created a hyperlink to a location in this workbook for each of the products in the second table.");
             });
         }
 
-        async function updateHyperlink() {
+        async function updateHyperlinks() {
```

Comment:
Not mandatory, but IMHO, this is essentially identical to the create* method. I don't think we need both. Can we just have a comment above the create* method that says something like "use this to create or update a hyperlink"?

Reply:
You're right, it just so happens that the process for creating a range hyperlink and updating a range hyperlink are identical.  Creating separate functions (and having separate buttons) for create and update was done per Sudhi's request, so probably best to leave it this way.